twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -35,8 +35,10 @@
      * Invoked when new messages is available.
      * @param messages Iterator of new messages. The {@link FetchedMessage} instance maybe reused in the Iterator
      *                 and across different invocation.
+     * @return A long larger than zero as the offset to restart fetching messages when offset error is caught,
```

Comment:
what happens if the offset returned is out of bounds? Should document the expected behavior in such circumstances. Looking at the implementation, it seems like we read from the earliest offset.